src/engine/GroupBy.cpp

```diff
@@ -383,23 +374,33 @@ ProtoResult GroupBy::computeResult(bool requestLaziness) {
   if (!subresult->isFullyMaterialized()) {
     AD_CORRECTNESS_CHECK(metadataForUnsequentialData.has_value());
 
-    auto localVocabPointer =
-        std::make_shared<LocalVocab>(std::move(localVocab));
-    cppcoro::generator<IdTable> generator = CALL_FIXED_SIZE(
+    cppcoro::generator<Result::IdTableVocabPair> generator = CALL_FIXED_SIZE(
         (std::array{inWidth, outWidth}), &GroupBy::computeResultLazily, this,
         std::move(subresult), std::move(aggregates),
         std::move(metadataForUnsequentialData).value().aggregateAliases_,
-        std::move(groupByCols), localVocabPointer, !requestLaziness);
+        std::move(groupByCols), !requestLaziness);
 
-    return requestLaziness
-               ? ProtoResult{std::move(generator), resultSortedOn(),
-                             std::move(localVocabPointer)}
-               : ProtoResult{cppcoro::getSingleElement(std::move(generator)),
-                             resultSortedOn(), std::move(*localVocabPointer)};
+    if (requestLaziness) {
+      return ProtoResult{std::move(generator), resultSortedOn()};
+    }
+    Result::IdTableVocabPair pair =
+        cppcoro::getSingleElement(std::move(generator));
+    return ProtoResult{std::move(pair.idTable_), resultSortedOn(),
+                       std::move(pair.localVocab_)};
   }
 
   AD_CORRECTNESS_CHECK(subresult->idTable().numColumns() == inWidth);
 
+  // Make a deep copy of the local vocab from `subresult` and then add to it (in
+  // case GROUP_CONCAT adds a new word or words).
+  //
+  // TODO: In most GROUP BY operations, nothing is added to the local
+  // vocabulary, so it would be more efficient to first share the pointer here
+  // (like with `shareLocalVocabFrom`) and only copy it when a new word is about
+  // to be added. Same for BIND.
+
```

Comment:
That TODO seems out of date. 
The shared semantics of the local vocab have no deep copy semantics.
So just write something like "make a copy of the local vocab...  Note: the LocalVocab has reference semantics via `shared_ptr`, so no actual strings are copied here.